### ISSUE TITLE
only_throw_errors: warn `throw Never`

### ISSUE THREAD

**Issue by untp** [dart-lang/linter#2886]
The `only_throw_errors` lint should warn when `throw Never` used.

**To Reproduce**
`throw Never;`

**Additional context**
Currently the lint warns other fancy throws like `throw Null;`, `throw dynamic;`, `throw (){};`.

**Comment by untp** [comment]
This is fixed on beta. I tested with these versions.
```
[✓] Flutter (Channel beta, 2.5.0-5.2.pre, on Microsoft Windows [Version 10.0.19041.1165], locale en-US)
    • Flutter version 2.5.0-5.2.pre at C:\Users\U\AppData\Local\Programs\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 19c61fed0d (4 days ago), 2021-08-18 17:10:31 -0700
    • Engine revision 7a4c4505f6
    • Dart version 2.14.0 (build 2.14.0-377.7.beta)
```
```
[✓] Flutter (Channel stable, 2.2.3, on Microsoft Windows [Version 10.0.19041.1165], locale en-US)
    • Flutter version 2.2.3 at C:\Users\U\AppData\Local\Programs\flutter
    • Framework revision f4abaa0735 (7 weeks ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4
```
